src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1964,6 +1970,27 @@ void signJar(String jarName, String alias)
 
         Throwable failedCause = null;
         String failedMessage = null;
+        JarFile asJar = null;
+
+        try {
+            asJar = new JarFile(jarFile);
+        } catch (IOException ioe) {
+            // intentionally "eat" this, since we don't want to fail, if we
+            // cannot perform the multiple manifest check to output the warning
+        }
+
+        // only perform this check if the ZipFile is a JarFile
+        if(asJar != null && verbose != null) {
+            int manifestNum = JUZFA.getManifestNum(asJar);
+            try {
+                asJar.close();
+            } catch (IOException ioe) {
+                // intentionally "eat" this, since we don't want to fail
+            }
+            if(manifestNum > 1) {
+                hasMultipleManifests = true;
+            }
+        }
```

Comment:
Can we put everything above in a try-with-resources block?

Also, are you sure we only print out this warning when `verbose != null`?

Reply:
I'll refactor to use try-with-resources.

It was originally described to me as a verbose message, but I don't see anything in the bug description to require it only during verbose. I've removed the check.